src/libraries/Microsoft.PowerFx.Core/Parser/ParserOptions.cs

```diff
@@ -20,18 +20,39 @@ public class ParserOptions
         /// </summary>
         public bool AllowsSideEffects { get; set; }
 
+        /// <summary>
+        /// The culture that an expression will parse with. 
+        /// This primarily determines numeric decimal separator character
+        /// as well as chaining operator. 
+        /// </summary>
         public CultureInfo Culture { get; set; }
 
+        /// <summary>
+        /// If greater than 0, enforces a maximum length on a single expression. 
+        /// It is an immediate parser error if the expression is too long. 
+        /// </summary>
+        public int MaxExpressionLength { get; set; }
+
         internal ParseResult Parse(string script)
         {
             return Parse(script, Features.None);
         }
 
         internal ParseResult Parse(string script, Features features)
         {
+            if (MaxExpressionLength > 0 && script.Length > MaxExpressionLength)
```

Comment:
>MaxExpressionLength > 0 [](http://example.com/codeflow?start=16&length=23)

This test should be in the setter

Reply:
>0 is just the "is enabled" check. 
It's whatever the host sets. If a host is running on a large dedicated VM   and wants to allow 100k strings, then they can configure to do that.